doc/cprover-manual/smt2-incr.md

```diff
@@ -0,0 +1,199 @@
+[CPROVER Manual TOC](../)
+
+## SMT 2 incremental backend
+
+CBMC supports many different smt solver backends. This document describes how to use the incremental
+smt2 solver. For other solver usages use `cbmc --help`.
+
+The incremental smt2 solver backend of CBMC is a solver-agnostic smt backend that allows the user to
+specify which smtlib 2.6 compliant solver CBMC should use. This choice increase the flexibility of
+CBMC as the user can easily switch solver at runtime allowing also custom parametrization of the
+solver to be used. Also the backend exploits the incremental mode of the smt solver, so subsequent
+incremental SMT solver queries do not require a full re-evaluation of the entire SMT formula.
+
+### Supported features
+
+The incremental smt2 solver supports the following CBMC C features:
+
+* Integral types. These are encoded into smt by using bit-vectors,
+* Pointers. Encoded by using smt bit-vectors,
+* Arrays. Encoded by using smt array theory.
+
+Currently unupported features include floating-point types, structs and unions.
+
+For more details on which features are supported see regression
+tests in [`regression/cbmc-incr-smt2`](../regression/cbmc-incr-smt2).
+
+### Requisites
+
+To run the incremental smt2 backend it is necessary to have an smt solver that:
+
+* is runnable from the command line,
+* supports interactive mode from the command line,
+* supports smtlib 2.6 input,
+* supports incremental mode.
+
+Because of the way supported features are encoded the smt solver should be capable to handle arrays,
+bit-vectors and uninterpreted functions and quantifiers. Also, the solver should work with the `ALL`
```

Comment:
bit-vectors and uninterpreted -> bit-vectors, uninterpreted

Reply:
changed